platforms/access-bridge/src/controllers/checkout-controller.ts

```diff
@@ -0,0 +1,65 @@
+import { StripeCheckoutParams } from '@jwp/ott-common/types/stripe.js';
+import { Request, Response, NextFunction } from 'express';
+
+import { StripeService } from '../services/stripe-service.js';
+import { AccessBridgeError, ErrorDefinitions, sendErrors } from '../errors.js';
+import { IdentityService } from '../services/identity-service.js';
+import { isValidSiteId } from '../utils.js';
+
+/**
+ * Controller class responsible for handling Stripe Checkout sessions.
+ */
+export class CheckoutController {
+  private identityService: IdentityService;
+  private stripeService: StripeService;
+
+  constructor() {
+    this.identityService = new IdentityService();
+    this.stripeService = new StripeService();
+  }
+
+  /**
+   * Service handler for initiating a Stripe Checkout session based on the provided price, mode and redirect URL.
+   * @param req - Express request object
+   * @param res - Express response object
+   * @param next - Express next middleware function
+   * @returns A Promise that resolves with a response containing the URL for the Stripe Checkout session.
+   */
+  async initiateCheckout(req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      const siteId = req.params.site_id;
+      if (!isValidSiteId(siteId)) {
+        sendErrors(res, ErrorDefinitions.ParameterInvalidError.create({ parameterName: 'site_id' }));
+        return;
+      }
+
+      const authorization = req.headers['authorization'];
+      if (!authorization) {
+        sendErrors(res, ErrorDefinitions.UnauthorizedError.create());
+        return;
+      }
+
+      const checkoutParams = req.body;
+
+      // Validate required params
+      const requiredParams: (keyof StripeCheckoutParams)[] = ['price_id', 'mode', 'success_url', 'cancel_url'];
```

Comment:
Should the urls be passed from the FE as params or should they be hard coded in the middleware?

Reply:
What urls you mean? The endpoints themselves? Why do we wanna hardcode them? what do we gain, when we have all the needed info on the FE. Plus the endpoints can be easily tested via API if the site_id is exposed in the route.